src/error.rs

```diff
@@ -181,16 +185,29 @@ impl Error {
     /// indicates a 4.4+ server, a label should only be added if the error is a network error.
     /// Otherwise, a label should be added if the error is a network error or the error code
     /// matches one of the retryable write codes.
-    pub(crate) fn should_add_retryable_write_label(&self, max_wire_version: i32) -> bool {
+    pub(crate) fn should_add_retryable_write_label(
+        &self,
+        max_wire_version: i32,
+        conn: Option<&Connection>,
```

Comment:
I'd probably pass in `Option<ServerType>` instead here, since that's all this needs and it'll avoid the need for the return to be `Result`.

Reply:
Done